crates/oxide/src/extractor/mod.rs

```diff
@@ -0,0 +1,902 @@
+            ),
+            // Quoted attribute
+            (r#"div class="px-2.5""#, vec!["div", "px-2.5"]),
+        ] {
+            assert_extract_sorted_candidates(&pre_process_input(input, "slim"), expected);
+        }
+    }
+
+    // C# syntax, see: https://github.com/tailwindlabs/tailwindcss/issues/16189#issue-2826350984
+    #[test]
+    fn test_csharp_syntax() {
+        for (input, expected) in [
+            // Not a valid arbitrary value or variant
+            // Extract inner candidates: `gap-y-4`
+            (r#"[CssClass("gap-y-4")]"#, vec!["gap-y-4"]),
+            (r#"[CssClass("hover:gap-y-4")]"#, vec!["hover:gap-y-4"]),
+            // Valid arbitrary variant + utility, extract full candidate without inner candidates
+            (
+                r#"[CssClass("gap-y-4")]:flex"#,
+                vec![r#"[CssClass("gap-y-4")]:flex"#],
+            ),
+        ] {
+            assert_extract_sorted_candidates(input, expected);
+        }
+    }
+
+    // Clojure syntax, see: https://github.com/tailwindlabs/tailwindcss/issues/16189#issuecomment-2642438176
+    #[test]
+    fn test_clojure_syntax() {
+        for (input, expected) in [
+            (r#"[:div {:class ["p-2"]}"#, vec!["p-2"]),
+            (
+                r#"[:div {:class ["p-2" "text-green"]}"#,
+                vec!["p-2", "text-green"],
+            ),
+            (r#"[:div {:class ["p-2""#, vec!["p-2"]),
+            (r#"               "text-green"]}"#, vec!["text-green"]),
+            (r#"[:div.p-2]"#, vec!["p-2"]),
+        ] {
+            assert_extract_sorted_candidates(input, expected);
+        }
+    }
+
+    // Gleam syntax, see: https://github.com/tailwindlabs/tailwindcss/issues/15632#issuecomment-2617431021
+    #[test]
+    fn test_gleam_syntax() {
+        for (input, expected) in [
+            (r#"html.div([attribute.class("py-10")], [])"#, vec!["py-10"]),
+            (
+                r#"html.div([attribute.class("hover:py-10")], [])"#,
+                vec!["hover:py-10"],
+            ),
+        ] {
+            assert_extract_sorted_candidates(input, expected);
+        }
+    }
+
+    #[test]
+    fn test_overlapping_candidates() {
+        for (input, expected) in [
+            // Not a valid arbitrary property
+            // Not a valid arbitrary variant
+            // Extract inner candidates
+            (
+                r#"[CssClass("[&:hover]:flex",'italic')]"#,
+                vec!["[&:hover]:flex", "italic"],
+            ),
+            // Not a valid arbitrary property or variant, array syntax
+            // Extract inner candidates
+            (
+                r#"["flex",["italic",["underline"]]]"#,
+                vec!["flex", "italic", "underline"],
+            ),
+            // Not a valid arbitrary variant (not followed by a candidate)
+            // Inner classes `is`, `italic` and `underline` are not valid in this context
+            (r#"[:is(italic):is(underline)]"#, vec![]),
```

Comment:
If `(` and `)` were considered valid boundary characters then this should extract `is`, `italic` and `underline` but don't think that's necessary 🤔

Reply:
I think this is okay. I can't think of a situation where `(…)` marks a class list. We can always update things if someone opens an issue about it.

Reply:
Only thing I can think of is Ruby, where you can also do `%w(flex underline)`, but if we want to support that I feel like it should be a preprocessing step as well.

Reply:
oh… I didn't realize that worked. Yeah I agree, make that a pre-processing step.